source/geod24/concurrency.d

```diff
@@ -1911,38 +1950,9 @@ private
          * Throws:
          *  An exception if the queue is full and onCrowdingDoThis throws.
          */
-        final void put(ref Message msg)
+        public bool put (ref Message msg)
         {
-            synchronized (m_lock)
-            {
-                // TODO: Generate an error here if m_closed is true, or maybe
-                //       put a message in the caller's queue?
-                if (!m_closed)
-                {
-                    while (true)
-                    {
-                        if (isPriorityMsg(msg))
-                        {
-                            m_sharedPty.put(msg);
-                            m_putMsg.notify();
-                            return;
-                        }
-                        if (!mboxFull() || isControlMsg(msg))
-                        {
-                            m_sharedBox.put(msg);
-                            m_putMsg.notify();
-                            return;
-                        }
-                        if (m_onMaxMsgs !is null && !m_onMaxMsgs(thisTid))
-                        {
-                            return;
-                        }
-                        m_putQueue++;
-                        m_notFull.wait();
-                        m_putQueue--;
-                    }
-                }
-            }
+            return this._put(msg);
```

Comment:
I don't see why `_put` and `put` need to be separate methods. All `put` does is calls `_put` with the same arguments. You can just keep all the code inside of `put` itself.